programmers/dart-game.js

```diff
@@ -0,0 +1,37 @@
+function solution(dartResult) {
+    const squares = dartResult.match(/[SDT]/g);
+    const parts = [];
+    let indexPoint;
+    let sum = 0;
+    for(let i = 0; i < 3 ; i++) {
+        indexPoint = dartResult.indexOf(squares[i]) // 기준점이 되는 SDT의 인덱스
+        if ( i < 2) {   // 마지막 문자는 비교할 뒤의 인덱스가 존재하지 않기 때문에 나눠줌
+            if (dartResult[indexPoint + 1] === "*" || dartResult[indexPoint + 1] === "#") {
+                parts.push(dartResult.substring(0,indexPoint + 2));
+                dartResult = dartResult.substring(indexPoint + 2, dartResult.length); //특수문자까지 포함해야 하므로 +2
+            } else {
+                parts.push(dartResult.substring(0,indexPoint+1));
+                dartResult = dartResult.substring(indexPoint+1,dartResult.length); // 자신까지만 포함해야 하므로 +1
+            }
+        } else parts.push(dartResult.substring(0,dartResult.length));
+    }          // 필수적으로 들어가는 SDT기준으로 뒤의 문자가 특수문자가 아니라면 그 인덱스 기준으로 잘라서 PART에 넣고 문자열을 재정의 하여 반복
+    for (let j = 0; j < 3 ; j++) {
+        let num = parseInt(parts[j]);
+        if (squares[j] === "D") num = Math.pow(num,2);   // SDT기준으로 제곱해줌
+        else if (squares[j] === "T") num = Math.pow(num,3);
+        if (/\*/.test(parts[j])) {
+            if (j >= 1) {
+                parts[j - 1] *= 2;
+                parts[j] = num * 2;
```

Comment:
정답을 맞추셔서 딱히 드릴 말씀이 없습니다.
허나, 컨벤션을 조금 더 생각하면서 푸시면 좋을거 같습니다.
깔끔하게 푼 다른 사람의 코드를 보는 것도 도움이 될 거 같아 올립니다.
```suggestion
function solution(dartResult) {
    const bonus = { 'S': 1, 'D': 2, 'T': 3 },
          options = { '*': 2, '#': -1, undefined: 1 };

    let darts = dartResult.match(/\d.?\D/g);

    for (let i = 0; i < darts.length; i++) {
        let split = darts[i].match(/(^\d{1,})(S|D|T)(\*|#)?/),
            score = Math.pow(split[1], bonus[split[2]]) * options[split[3]];

        if (split[3] === '*' && darts[i - 1]) darts[i - 1] *= options['*'];

        darts[i] = score;
    }
    return darts.reduce((a, b) => a + b);
}
```

Reply:
감사합니다. 참고하여 컨벤션에 더 신경쓰겠습니다..8ㅅ8 죄송합니다.